SOLIDITY_STYLE_GUIDE.md

```diff
@@ -500,14 +501,21 @@ Hence, functions should be grouped according to their visibility and ordered:
 - internal functions
 - private functions
 
+The suggested order for variables' declarations is:
+
+- public variables
+- internal variables
+- internal variables
+
 Within a grouping, place the `view` and `pure` functions last.
 
 To read contract top down the following order is suggested:
 
-- using
-- event
-- enum
-- struct
+- usings
+- events
+- constants
```

Comment:
Is there a reason why the constants come after the events? I am used to having constants at the top of everything. Though I don't know why. This is also good, just wanted to know if there is a specific reason.

Reply:
Events naturally seems to be an external interface.
Constants could also include private, internal. That's why put it as high as possible in order, only after events and usings :).